pkg/graph/resolvers/base_struct.go

```diff
@@ -1,20 +1,46 @@
 package resolvers
 
 import (
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/mint-app/pkg/authentication"
 	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/storage"
 )
 
 // BaseStructPreUpdate applies incoming changes from to a TaskList Section, and validates it's status
-func BaseStructPreUpdate(bs models.IBaseStruct, changes map[string]interface{}, principal string) error {
+func BaseStructPreUpdate(logger *zap.Logger, bs models.IBaseStruct, changes map[string]interface{}, principal authentication.Principal, store *storage.Store, applyChanges bool) error {
```

Comment:
Is there a reason this would be on `BaseStructPreUpdate` and take an `IBaseStruct`? I would assume that we'd want to pass something that specifically has a ModelPlanID 🤔

Reply:
The reason I put the check there is because all objects should be going through BaseStructPreUpdate, since every object should be a BaseStruct. 

There are edge cases where you want to check something that doesn't have a ModelPlanID directly, so you need to handle it by other parameters (eg discussion id). I tried an implementation where I just wrote that logic directly in the separate resolver, but I didn't like it as much as this. A structure like this means that as long as it is a base struct, access will be checked.

```mermaid
flowchart LR
a1(PreUpdate)
a1--> a2(Check If Struct Requires Access Check)
a2 -->a3(Yes) 
a3-->a10(AssessmentUser?)
a10--True-->a5
a10-->a11(False)
a11--ModelPlanRelation--> a6(CheckByModelPlanID)
a11--DiscussionRelation-->a7(CheckByDiscussionID)
a6-->a8(Collaborator?)
a7-->a8
a8--True-->a5
a8--false-->a9(ReturnError)
a2 --> a4(No)
a4-->a5(ContinueWithAction)


```